webapp-django/crashstats/crashstats/decorators.py

```diff
@@ -43,7 +46,17 @@ def pass_default_context(view):
     def inner(request, *args, **kwargs):
         product = kwargs.get('product', None)
         versions = kwargs.get('versions', None)
-        kwargs['default_context'] = utils.build_default_context(product,
-                                                                versions)
+        user_authenticated = request.user.is_authenticated()
+        try:
+            kwargs['default_context'] = utils.build_default_context(
+                product,
+                versions,
+                user_authenticated
+            )
+        except utils.MissingProductError:
+            return redirect(
+                reverse('manage:products') + '?' +
+                urllib.urlencode({'product': product})
```

Comment:
You don't need to build that URL yourself. Take a look at the examples here: https://docs.djangoproject.com/en/1.6/topics/http/shortcuts/#examples You can do something like ``redirect('url', product=product)``. This way you don't need to use urllib, and it's easier to maintain.